src/selectors.ts

```diff
@@ -184,9 +189,10 @@ export class Selectors {
       if (eqIndex !== -1 && part.substring(0, eqIndex).trim().match(/^[a-zA-Z_0-9-+:]+$/)) {
         name = part.substring(0, eqIndex).trim();
         body = part.substring(eqIndex + 1);
-      } else if (part.startsWith('"')) {
+      } else if (hasTextSelectorSurroundings(part)) {
         name = 'text';
-        body = part;
+        const innerText = part.slice(1, part.length - 1);
+        body = innerText.includes('\\') ? JSON.parse(`"${innerText}"`) : innerText;
```

Comment:
Note this is not exactly equivalent, since it turns strict notation to the relaxed one by getting rid of the quotes. I was thinking about trying the JSON.parse to only check for the error earlier.